### PR TITLE
Feature/community/post trade

### PR DESCRIPTION
**Pull request by almujtabamohammed**
## Related to
[647](https://techxarya.atlassian.net/browse/CRW-647?atlOrigin=eyJpIjoiYzY2ZTlkYjEzMGRiNGRkOWFiZWJhM2RkMjFlYWI0YTAiLCJwIjoiaiJ9)

### CODE REVIEW

File: src/components/app/community/Post/PostContent/PostFinancialData/FinancialDataPopup/FinancialDataPopup.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            setLoading(false);
+        });
+    }, [getPostExpirationDateOffset, isPostActive, post?.FinancialData?.Instrument?.MarketItem, post?.FinancialData?.Timeframe, postCreationTime]);
+
+    const ideaChart = useMemo(() => {
+        const marketItem = post?.FinancialData?.Instrument?.MarketItem;
+        const hasFinancialData = post?.FinancialData?.EntryValue || post?.FinancialData?.BreakEven || post?.FinancialData?.StopLoss || post?.FinancialData?.TakeProfits;
+
+        if (marketItem && postCreationTime && hasFinancialData && post?.FinancialData) {
+            console.log({ chartData })
```
Comment (AbdallaEmadEldinAbdalla) [inline src/components/app/community/Post/PostContent/PostFinancialData/FinancialDataPopup/FinancialDataPopup.tsx:95]: Delete console

File: src/components/shared/charts/graph/candlesticksChart.tsx [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+                            position,
+                            color,
+                            shape,
+                            text: `${pricePointsShortForm[priceEvaluation.Type]} @ ` + shortNumberFormat(priceEvaluation.Price)
+                        });
+                    }
+                });
+            }
+
+            console.log({ markers }, financialData.PriceEvaluations)
```
Comment (AbdallaEmadEldinAbdalla) [inline src/components/shared/charts/graph/candlesticksChart.tsx:215]: Delete console

File: src/components/shared/charts/graph/candlesticksChart.tsx [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            };
+        },
+        [primaryData, financialData, postCreationTime, isMobileOnly, theme, fixed, manageTooltip]
+    );
+
+    const loadingOverlay = useMemo(() => {
+        if (isLoading) {
+            return (
+                <Col className="items-center justify-center bg-offWhite-3 dark:bg-[#00000080] absolute w-full h-full z-10">
+                    <p className="text-2xl font-bold">Calculating...</p>
```
Comment (AbdallaEmadEldinAbdalla) [inline src/components/shared/charts/graph/candlesticksChart.tsx:297]: Translate

File: src/services/controllers/asset.tsx [hunk 4]
```diff
@@ -91,7 +91,7 @@ export const getAssetTimeseriesPrice = async (
 };
 
 export const getAssetSparkLineData = async (symbol: string) => {
-  return await axios.get(`${process.env.NEXT_PUBLIC_TWELEVE_API_URL}?symbol=${symbol}/usd&interval=1h&outputsize=168&apikey=${process.env.NEXT_PUBLIC_TWELVE_DATA_API_KEY}&timezone=UTC`)
+  return await axios.get(`${process.env.NEXT_PUBLIC_TWELEVE_API_URL}?symbol=${symbol}/usd&interval=8h&outputsize=21&apikey=${process.env.NEXT_PUBLIC_TWELVE_DATA_API_KEY}&timezone=UTC`)
```
Comment (AbdallaEmadEldinAbdalla) [inline src/services/controllers/asset.tsx:94]: Revert this change it ruining the community search and market asset page result

**Review by AbdallaEmadEldinAbdalla — CHANGES_REQUESTED** [review]

File: src/components/app/community/Post/PostContent/PostFinancialData/FinancialDataPopup/FinancialDataPopup.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+                    />
+                </Row>
+            )
+        }
+    }, [chartData, isLoading, post?.FinancialData, postCreationTime]);
+
+
+    return (
+        show ? <Modal
```
Comment (AbdallaEmadEldinAbdalla) [inline src/components/app/community/Post/PostContent/PostFinancialData/FinancialDataPopup/FinancialDataPopup.tsx:126]: Why `show` is outside! xD

**Review by AbdallaEmadEldinAbdalla — CHANGES_REQUESTED** [review]

File: src/components/app/community/Post/PostContent/PostFinancialData/FinancialDataPopup/FinancialDataPopup.tsx (hunk 5 above)
Comment (almujtabamohammed) [reply src/components/app/community/Post/PostContent/PostFinancialData/FinancialDataPopup/FinancialDataPopup.tsx:126]: To reset the state inside the Modal

**Pull request merged**